### PR TITLE
Prepare for re-enabled opcodes

### PR DESCRIPTION
**Pull request by dagurval**
Prepare for re-enabled opcodes for the May 2018 protocol upgrade - https://reviews.bitcoinabc.org/D1141
Add a flag to gate opcodes activation for the new HF - https://reviews.bitcoinabc.org/D1208
Add helper function for disabled opcodes - https://reviews.bitcoinabc.org/D1209

### CODE REVIEW

File: src/core_read.cpp [hunk 1]
```diff
@@ -30,9 +30,9 @@ CScript ParseScript(std::string s)
 
     if (mapOpNames.empty())
     {
-        for (int op = 0; op <= OP_NOP10; op++)
+        for (int op = 0; op <= FIRST_UNDEFINED_OP_VALUE; op++)
```
Comment (dgenr8) [inline src/core_read.cpp]: Why <= rather than <?  This changes the semantics.

Comment (Danconnolly) [reply src/core_read.cpp]: Yes, I'm not sure how this happened. In https://reviews.bitcoinabc.org/D1141 it is "<".

Comment (dagurval) [reply src/core_read.cpp]: Good eye @dgenr8. 

That’s a bad merge by me. ABC reformatted all the files so cherry picking involves a lot of manual merging.

**Comment by dagurval** [comment]
Fixed issue found by @dgenr8

**Pull request merged**